tests/lib/nested.sh

```diff
@@ -1288,25 +1283,24 @@ nested_start_core_vm_unit() {
             mv OVMF_VARS.ms.fd /usr/share/OVMF/OVMF_VARS.ms.fd
         fi
 
-        # In this case the kernel.efi is unsigned and signed with snaleoil certs
-        if [ "$NESTED_FORCE_MS_KEYS" != "true" ] && [ "$NESTED_BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
-            OVMF_VARS="snakeoil"
-        else
-            OVMF_VARS="ms"
-        fi
-
         if nested_is_secure_boot_enabled; then
-            OVMF_CODE="secboot"
-            if os.query is-arm; then
-                cp -f "/usr/share/AAVMF/AAVMF_VARS.fd" "$NESTED_ASSETS_DIR/AAVMF_VARS.fd"
-                PARAM_BIOS="-drive file=/usr/share/AAVMF/AAVMF_CODE.fd,if=pflash,format=raw,unit=0,readonly=on -drive file=$NESTED_ASSETS_DIR/AAVMF_VARS.fd,if=pflash,format=raw"
+            OVMF_CODE=".secboot"
+            if [ "$NESTED_FORCE_MS_KEYS" != "true" ] && [ "$NESTED_BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
+                OVMF_VARS=".snakeoil"
             else
-                cp -f "/usr/share/OVMF/OVMF_VARS.${OVMF_VARS}.fd" "$NESTED_ASSETS_DIR/OVMF_VARS.${OVMF_VARS}.fd"
-                PARAM_BIOS="-drive file=/usr/share/OVMF/OVMF_CODE.${OVMF_CODE}.fd,if=pflash,format=raw,unit=0,readonly=on -drive file=$NESTED_ASSETS_DIR/OVMF_VARS.${OVMF_VARS}.fd,if=pflash,format=raw"
-                PARAM_MACHINE="-machine q35${ATTR_KVM} -global ICH9-LPC.disable_s3=1"
+                OVMF_VARS=".ms"
             fi
         fi
 
+        if os.query is-arm; then
+            cp -f "/usr/share/AAVMF/AAVMF_VARS.fd" "$NESTED_ASSETS_DIR/AAVMF_VARS.fd"
+            PARAM_BIOS="-drive file=/usr/share/AAVMF/AAVMF_CODE.fd,if=pflash,format=raw,unit=0,readonly=on -drive file=$NESTED_ASSETS_DIR/AAVMF_VARS.fd,if=pflash,format=raw"
+        else
+            cp -f "/usr/share/OVMF/OVMF_VARS${OVMF_VARS}.fd" "$NESTED_ASSETS_DIR/OVMF_VARS${OVMF_VARS}.fd"
```

Comment:
OVMF_VARS can be unset, it's only set in `if nested_is_secure_boot_enabled; then`

Reply:
from nested tests:
```
 + tests.nested create-vm core
'/var/tmp/work-dir/images/ubuntu-core-24-custom-kernel-modules-components_plain-pc.img' -> '/var/tmp/work-dir/images/ubuntu-core-current.img'
cp: cannot stat '/usr/share/OVMF/OVMF_VARS.fd': No such file or directory
-----
.
2025-01-28 18:54:36 Preparing google-nested:ubuntu-24.04-64:tests/nested/manual/uc20-install-in-initrd:hook (jan281838-980696)...
2025-01-28 18:54:36 Preparing google-nested:ubuntu-24.04-64:tests/nested/manual/split-refresh (jan281838-980686)...
2025-01-28 18:54:37 Debug output for google-nested:ubuntu-24.04-64:tests/nested/manual/kernel-modules-components:plain (jan281838-980720) : saved to file spread-logs/google-nested_ubuntu-24.04-64_tests_nested_manual_kernel-modules-components.debug.log
```

Reply:
That was the point. The file should have existed. It was for non-secboot.

Reply:
It is OVMF_VARS_4M.fd...